handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -248,6 +250,39 @@ protected void deallocate() {
         }
     }
 
+    /**
+     * Enables OCSP stapling on the {@link SSLEngine} and throws an {@link Exception} if OCSP stapling is not supported.
+     */
+    @UnstableApi
+    public void enableOcsp(ReferenceCountedOpenSslContext context) {
+        context.enableOcsp();
```

Comment:
Hum this relationship is unnatural. Typically the `SSlContext` provides the default configuration for creating an `SSLEngine` and the same relationship olds for `SSL_CTX` and `SSL`. Could we just instead have tcnative throw if the `SSL_CTX` has not been setup correctly? For example we could add a `bool oscp_enabled` to `tcn_ssl_ctxt_t` (in `ssl_private.h`). `SSLContext#enableOcsp` would set this state (even for `BoringSSL`) and then `SSL#enableOcsp` would check this state has been set. You can get the `tcn_ssl_ctxt_t` from the `SSL*` via `SSL_get_app_data2`.